File: scratch/test_decklog.dart
import 'dart:io';
import 'dart:convert';

void main() async {
  final client = HttpClient();
  
  // Test POST system/en/api/view/4DWCZ
  await testUrl(client, 'POST', 'https://decklog-en.bushiroad.com/system/en/api/view/4DWCZ');
  // Test GET system/en/api/view/4DWCZ
  await testUrl(client, 'GET', 'https://decklog-en.bushiroad.com/system/en/api/view/4DWCZ');
  // Test POST system/ja/api/view/4DWCZ
  await testUrl(client, 'POST', 'https://decklog-en.bushiroad.com/system/ja/api/view/4DWCZ');
  // Test GET system/ja/api/view/4DWCZ
  await testUrl(client, 'GET', 'https://decklog-en.bushiroad.com/system/ja/api/view/4DWCZ');

  client.close();
}

Future<void> testUrl(HttpClient client, String method, String url) async {
  print('=== Testing $method on $url ===');
  try {
    final uri = Uri.parse(url);
    final request = await client.openUrl(method, uri);
    request.headers.set('accept', 'application/json, text/plain, */*');
    request.headers.set('content-type', 'application/json;charset=UTF-8');
    
    // Send empty payload for POST
    if (method == 'POST') {
      request.write(jsonEncode({}));
    }
    
    final response = await request.close();
    print('Status: ${response.statusCode}');
    final body = await response.transform(utf8.decoder).join();
    print('Response length: ${body.length}');
    if (body.length > 500) {
      print(body.substring(0, 500));
    } else {
      print(body);
    }
  } catch (e) {
    print('Error: $e');
  }
}

File: scratch/test_decklog.py
import requests
import json

print("=== Hitting system/en/api/view/4DWCZ with POST ===")
try:
    r = requests.post("https://decklog-en.bushiroad.com/system/en/api/view/4DWCZ")
    print(f"Status: {r.status_code}")
    print(r.text[:500])
except Exception as e:
    print(f"Error: {e}")

print("=== Hitting system/en/api/view/4DWCZ with GET ===")
try:
    r = requests.get("https://decklog-en.bushiroad.com/system/en/api/view/4DWCZ")
    print(f"Status: {r.status_code}")
    print(r.text[:500])
except Exception as e:
    print(f"Error: {e}")

print("=== Hitting system/en/api/view with POST payload ===")
try:
    r = requests.post("https://decklog-en.bushiroad.com/system/en/api/view", json={"code": "4DWCZ"})
    print(f"Status: {r.status_code}")
    print(r.text[:500])
except Exception as e:
    print(f"Error: {e}")

print("=== Hitting system/ja/api/view/4DWCZ with POST ===")
try:
    r = requests.post("https://decklog-en.bushiroad.com/system/ja/api/view/4DWCZ")
    print(f"Status: {r.status_code}")
    print(r.text[:500])
except Exception as e:
    print(f"Error: {e}")
